doc/developers-guide/message_routing.md

```diff
@@ -1,59 +1,49 @@
+The stanza is checked against any privacy lists in use and, in the case of being allowed, it will be routed by `ejabberd_router:route/3`. 
+This also takes into account "blocking commands", which are part of the privacy system.
 
 ## 4. Chain of routing
 
-Further on, behaviour is configurable: `ejabberd_router:route/3` passes the stanza through
-a chain of routing modules and applying `Mod:filter/3` and `Mod:route/3` from each of them.
+Further on, the behaviour is configurable: `ejabberd_router:route/3` passes the stanza through a chain of routing modules and applies `Mod:filter/3` and `Mod:route/3` from each of them.
 Each of those modules has to implement `xmpp_router` behaviour.
 
 A module can choose to either:
 
-* drop the stanza, or route the stanza
-* pass the stanza on unchanged, or modify the stanza and pass the result on
+* drop/route the stanza,
+* pass the stanza on unchanged/modify the stanza and pass the result on.
 
-The set of routing modules can be set in configuration as `routing_modules`. The default
-behaviour is the following:
+A set of routing modules can be set in configuration as `routing_modules`. 
+The default behaviour is the following:
 
 * `mongoose_router_global`: runs a global `filter_packet` hook
-* `mongoose_router_external_local`: checks if there is an external component registered for the
-destination domain on the node we are on now, possibly routes the stanza to it
-* `mongoose_router_external`: checks if there is an external component registered for the
-destination domain on any node in the cluster, possibly routes the stanza to it
-* `mongoose_router_localdomain`: checks if there is a local route registered for the destination
-domain (i.e. there is an entry in mnesia `route` table), possibly routes the stanza to it
+* `mongoose_router_external_local`: checks if there is an external component registered for the destination domain on the current node, possibly routes the stanza to it
+* `mongoose_router_external`: checks if there is an external component registered for the destination domain on any node in the cluster, possibly routes the stanza to it
+* `mongoose_router_localdomain`: checks if there is a local route registered for the destination domain (i.e. there is an entry in mnesia `route` table), possibly routes the stanza to it
 * `ejabberd_s2s`: tries to find or establish a connection to another server and send the stanza there
 
 ![You should see an image here; if you don't, use plantuml to generate it from routing.uml](routing.png)
 
 ## 5. Look up `external_component` and `route`
 
-An external component or local route is obtained by looking up `external_component` and `route`
-mnesia tables, respectively. Whats in there is either a fun to call or an MF to apply:
+An external component and a local route are obtained by looking up `external_component` and `route` mnesia tables, respectively. Whats in there is either a fun to call or an MF to apply:
```

Comment:
Whats => What's

We could rewrite this sentence to something like "Items in these tables include funs or MFs to apply."